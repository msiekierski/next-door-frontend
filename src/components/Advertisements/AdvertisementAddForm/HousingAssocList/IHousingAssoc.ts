export interface IHousingAssoc {
  idAssoc: number;
  city: string;
  street: string;
  postalCode: string;
  name: string;
}
