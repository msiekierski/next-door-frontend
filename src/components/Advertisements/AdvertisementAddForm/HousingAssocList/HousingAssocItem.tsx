import React, { ChangeEvent, FunctionComponent } from "react";
import { Container, Form, ListGroup } from "react-bootstrap";
import { IHousingAssoc } from "./IHousingAssoc";
import { ADVERT_BOTTOM_PAYMENT } from "../../../../constants/constants";

type Props = IHousingAssoc & {
  setChecked: Function;
  setCharge: Function;
  setIsValid: Function;
};

const HousingAssocItem: FunctionComponent<Props> = ({ name, street, setChecked, idAssoc, setCharge, setIsValid }) => {
  function handleChecked(e: ChangeEvent<HTMLInputElement>) {
    if (e.target.checked) {
      setChecked((checked: number[]) => {
        const newChecked: number[] = [...checked, idAssoc];
        setCharge(100 * newChecked.length);
        if (newChecked.length >= 1) {
          setIsValid(true);
        }
        return newChecked;
      });
    } else {
      setChecked((checked: number[]) => {
        const newChecked: number[] = checked.filter((e) => e != idAssoc);
        setCharge(100 * newChecked.length);
        if (newChecked.length == 0) {
          setIsValid(false);
        }
        return newChecked;
      });
    }
  }

  return (
    <ListGroup.Item className={"pl-0 pr-0"}>
      <Container className={"d-flex justify-content-between"}>
        <Form.Label className={`mb-0`}>
          {name}, {street}
        </Form.Label>
        <Form.Check type="checkbox" onChange={handleChecked} />
      </Container>
    </ListGroup.Item>
  );
};

export default HousingAssocItem;
