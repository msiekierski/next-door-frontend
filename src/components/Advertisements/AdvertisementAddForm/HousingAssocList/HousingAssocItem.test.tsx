import React from "react"
import renderer from "react-test-renderer"
import HousingAssocItem from "./HousingAssocItem"
import { IHousingAssoc } from "./IHousingAssoc";

type Props = IHousingAssoc & {
  setCharge: Function;
};


const props: Props = {
    name: "My Assoc",
    idAssoc: 1,
    city: "Wroclaw",
    street: "Rynek",
    postalCode: "50-000",
    setCharge: () => {}
}



it("renders correctly", () => {
  const tree = renderer.create(<HousingAssocItem {...props}/>);
  expect(tree).toMatchSnapshot();
});