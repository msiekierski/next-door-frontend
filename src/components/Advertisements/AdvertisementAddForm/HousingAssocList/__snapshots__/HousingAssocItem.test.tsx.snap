// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="pl-0 pr-0 list-group-item"
  disabled={false}
  onClick={[Function]}
>
  <div
    className="d-flex justify-content-between container"
  >
    <label
      className="mb-0 form-label"
    >
      My Assoc
      , 
      Rynek
    </label>
    <div
      className="form-check"
    >
      <input
        className="form-check-input position-static"
        disabled={false}
        onChange={[Function]}
        type="checkbox"
      />
    </div>
  </div>
</div>
`;
